Source/WebCore/platform/graphics/Font.h

```diff
@@ -425,6 +431,56 @@ ALWAYS_INLINE FloatRect Font::boundsForGlyph(Glyph glyph) const
     return bounds;
 }
 
+#if USE(CORE_TEXT)
+ALWAYS_INLINE Vector<FloatRect> Font::boundsForGlyphs(std::span<const Glyph> glyphs) const
+{
+    Vector<FloatRect> glyphBounds(glyphs.size(), FloatRect());
+
+    Vector<Glyph> glyphsNeedingMeasurement;
+    glyphsNeedingMeasurement.reserveInitialCapacity(glyphs.size());
+
+    Vector<size_t> positionsNeedingMeasurement;
```

Comment:
What if you do something like this to cut down on bookkeeping Vectors?

```C++
struct GlyphAndPosition {
    Glyph glyph;
    size_t position;
};

Vector<GlyphAndPosition> glyphsNeedingMeasurement;
```


You could also use a std::pair. but I like named field lookups rather than `foo.first` and `foo.second`.

Reply:
I'll give it a try!